vendor/woocommerce/action-scheduler/classes/ActionScheduler_ActionFactory.php

```
@@ -0,0 +1,378 @@
+	 * @return string The ID of the stored action
+	 * @throws InvalidArgumentException If $action is not a recurring action.
+	 */
+	public function repeat( $action ) {
+		$schedule = $action->get_schedule();
+		$next     = $schedule->get_next( as_get_datetime_object() );
+
+		if ( is_null( $next ) || ! $schedule->is_recurring() ) {
+			throw new InvalidArgumentException( __( 'Invalid action - must be a recurring action.', 'action-scheduler' ) );
+		}
+
+		$schedule_class = get_class( $schedule );
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$schedule_class'. (undefined)

(UnusedLocalVariable)